src/ptf/mask.py

```diff
@@ -46,7 +50,9 @@ def is_valid(self):
     def pkt_match(self, pkt):
         # just to be on the safe side
         pkt = str(pkt)
-        if len(pkt) != self.size:
+        # we compare up to the expected size, and fail if we haven't
+        # received enough bits
+        if len(pkt) < self.size:
```

Comment:
Why is it "desirable to compare only the prefix given by the mask.size"? I would argue that it is desirable to keep the current behavior which is: by default, unless explicitly requested by calling `set_do_not_care`, the match will fail if the received packet is not exactly the same as the requested packet.
I'm however completely in favor of enhancing the interface of the `Mask` class, by either adding an `ignore_extra_bytes` method or adding an optional parameter to the `__init__` method.

Reply:
added both an optional `__init__` parameter and the method.